content/en/docs/concepts/services-networking/gateway.md

```diff
@@ -0,0 +1,186 @@
+  * __Cluster Operator:__ Manages clusters and is typically concerned with policies, network
+    access, application permissions, etc.
+  * __Application Developer:__ Manages an application running in a cluster and is typically
+    concerned with application-level configuration and [Service][service] composition.
+* __Portable:__ Gateway API specifications are defined as [custom resources][crds] and are supported
+  by many [implementations][implementations].
+* __Expressive:__ Gateway API kinds support functionality for common traffic routing use cases
+  such as header-based matching, traffic weighting, and others that were only possible in [Ingress][ingress]
+  by using custom annotations.
+* __Extensible:__ Gateway allows for custom resources to be linked at various layers of the API.
+  This makes granular customization possible at the appropriate places within the API structure.
+
+Refer to the full Gateway API [documentation][gw-api] for additional details of these principles.
+
+## Resource model
+
+The Gateway API has three stable API kinds:
+
+* __GatewayClass:__ Defines a set of gateways with common configuration and managed by a controller
+  that implements the class.
+
+* __Gateway:__ Defines an instance of traffic handling infrastructure, such as cloud load balancer.
+
+* Route kinds: The Gateway API defines API kinds, HTTPRoute for example, that define protocol-specific
+  rules for mapping traffic from a Gateway listener to a representation of backend network endpoints.
+  These endpoints are often represented as a {{<glossary_tooltip text="Service" term_id="service">}}.
+
+The Gateway API is organized into different API kinds that have interdependent relationships to support
+the role-oriented nature of organizations. A Gateway object is associated with exactly one GatewayClass;
+the GatewayClass describes the gateway controller responsible for managing Gateways of this class.
+Route kinds such as HTTPRoute, are then associated to Gateways. A Gateway can filter the routes that
+may be attached to its `listeners`, forming a bidirection trust model with routes.
+
+The following figure illustrates the relationships of the three primary Gateway API resources:
+
+{{< figure src="/docs/images/gateway-kind-relationships.svg" alt="gateway-kind-relationships" class="diagram-medium" link="https://raw.githubusercontent.com/kubernetes/website/main/content/en/docs/images/gateway-kind-relationships.svg" >}}
+
+### GatewayClass {#api-kind-gateway-class}
+
+{{< gateway-feature-state for_gateway_version="v1.0" state="stable" >}}
+
+Gateways can be implemented by different controllers, often with different configurations. A Gateway
+must reference a GatewayClass that contains the name of the controller which implements the
+class. The GatewayClass can also define optional configuration parameters.
+
+A minimal GatewayClass example:
+
+{{% code_sample file="service/networking/gateway-class.yaml" %}}
```

Comment:
Something to watch out for: we test that our examples work (manually), and right now we try them with vanilla Kubernetes.

We'll have to work out how to support manifest examples where you need to install an extension, even an in-project extension, before they work.

It might be easier to use a YAML snippet and put a comment in the source to explain why. Then track an issue, post-merge, to track adding the support so we can use and test examples.
In parallel, we're looking to move that testing forward so that PRs fail CI unless the example is a valid manifest.

Reply:
I moved the examples to inline yaml snippets.